### PR TITLE
Work on the style and usability of find and replace...

### PR DESCRIPTION
**Pull request by ellisonbg**
* Using the align-left logo for the find in selection. We looked through various UIs and got this idea from Visual Studio Code. Still not perfect, but @jdfreder and I think it is an improvement.
* Small size input fields and buttons for more efficient usage of space.
* Bits of code organization.

### CODE REVIEW

**Comment by ellisonbg** [comment]
* Uses bold text for `Aa` and `.*` to improve legibility.

closes #558

File: notebook/static/notebook/js/searchandreplace.js [hunk 1]
```diff
@@ -147,58 +147,54 @@ define(function(require){
...
       .attr('id', 'isreg')
-      .addClass("btn btn-default")
+      .addClass("btn btn-default btn-sm")
       .attr('data-toggle','button')
+      .css('font-weight', 'bold')
       .attr('title', 'Use regex (JavaScript regex syntax)')
       .text('.*');
 
-    var onlySelectedButton = $('<button/>')
+    var onlySelectedButton = $('<button><i class="fa fa-align-left"/></button>')
```
Comment (Carreau) [inline notebook/static/notebook/js/searchandreplace.js]: doing that with jquery is bad. `$()` is not ment for templating, and it's inconsistent will al the rest of our code, like we won't be able to  for an `grep addClass | grep fa-align`

File: notebook/static/notebook/js/searchandreplace.js [hunk 2]
```diff
@@ -147,58 +147,54 @@ define(function(require){
...
       .attr('type', 'button')
-      .addClass("btn btn-default")
-      .append($('<i/>')
-        .addClass("fa fa-check-square-o")
-      )
+      .addClass("btn btn-default btn-sm")
       .attr('data-toggle','button')
-      .attr('title', 'Replace only in selected cells');
+      .attr('title', 'Find in selected cells');
```
Comment (Carreau) [inline notebook/static/notebook/js/searchandreplace.js]: I would prefer replace, Find and replace does not have the "Find" functionality, if we put Find only people will expect using this dialog to jump to the position where the match is found.

File: notebook/static/notebook/js/searchandreplace.js [hunk 3]
```diff
@@ -147,58 +147,54 @@ define(function(require){
...
-    var body = $('<div/>')
-      .attr('id', 'replace-preview');
-    var form = $('<form/>')
-      .attr('id', 'find-and-replace')
-      .append($('<div/>').addClass('form-group')
+      .attr('title', 'Match case')
+      .css('font-weight', 'bold')
+      .text('Aa');
+     
+    var search  = $("<input/>").addClass('form-control input-sm').attr('placeholder','Find');
```
Comment (Carreau) [inline notebook/static/notebook/js/searchandreplace.js]: line too long.

File: notebook/static/notebook/js/searchandreplace.js [hunk 4]
```diff
@@ -147,58 +147,54 @@ define(function(require){
...
         )
-      )
-      .append($('<div/>').addClass('form-group')
-        .append(repl)
-      )
-      .append(body);
+        .append(search)
+    )
 
+    var replace = $("<input/>").addClass('form-control input-sm').attr('placeholder','Replace');
```
Comment (Carreau) [inline notebook/static/notebook/js/searchandreplace.js]: line too long.

**Comment by Carreau** [comment]
Thanks for the cleanup.

**Comment by Carreau** [comment]
For more compactness :+1: 
```
pre.replace {
    padding:3px
}
```

Thoughts ?

File: notebook/static/notebook/js/searchandreplace.js (hunk 1 above)
Comment (ellisonbg) [inline notebook/static/notebook/js/searchandreplace.js]: Yep, fixed...

File: notebook/static/notebook/js/searchandreplace.js (hunk 2 above)
Comment (ellisonbg) [inline notebook/static/notebook/js/searchandreplace.js]: OK sounds good.

File: notebook/static/notebook/js/searchandreplace.js (hunk 3 above)
Comment (ellisonbg) [inline notebook/static/notebook/js/searchandreplace.js]: Fixed.

File: notebook/static/notebook/js/searchandreplace.js (hunk 4 above)
Comment (ellisonbg) [inline notebook/static/notebook/js/searchandreplace.js]: Fixed.

**Comment by ellisonbg** [comment]
On the padding. Using 3px seemed a bit tight, but here is a padding of `5px 10px` which matches that used in the small bootstrap input fields on the dialog:

screen shot 2015-11-24 at 5 55 51 pm

I think this looks good and definitely fits more than before.

**Comment by ellisonbg** [comment]
OK all changes are pushed and should be ready to go...

**Comment by minrk** [comment]
That's looking pretty good to me. I'll let @Carreau merge.

**Comment by Carreau** [comment]
+1 thanks ! Happy to see you submitting PRs !

**Comment by Carreau** [comment]
I'm thinking in using lighter red and green for match and delete. Thoughts?

**Comment by ellisonbg** [comment]
Sure I will try that now...

On Wed, Nov 25, 2015 at 4:31 AM, Matthias Bussonnier <
notifications@github.com> wrote:

> I'm thinking in using lighter red and green for match and delete. Thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/pull/741#issuecomment-159593735>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by ellisonbg** [comment]
Opps, did I merge this? Didn't mean to do that. But everything other than the colors is final so I think we are OK. I will create a seaprate PR for the color changes. After playing, I do like lighter colors.

**Comment by Carreau** [comment]
> Opps, did I merge this?

No I did, the color was a suggestion for separate PR.

**Comment by ellisonbg** [comment]
Ahh OK, other PR is submitted :)

On Wed, Nov 25, 2015 at 9:25 AM, Matthias Bussonnier <
notifications@github.com> wrote:

> Opps, did I merge this?
>
> No I did, the color was a suggestion for separate PR.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/pull/741#issuecomment-159680295>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Pull request merged**